### PR TITLE
Escape single quotes in string literals (SQL injection fix)

### PR DESCRIPTION
**Pull request by lukevanin**
## Summary

`XLiteFormatter.text(_:)` emitted string values verbatim inside single quotes with no escaping. Any string containing a `'` produced broken SQL, and attacker-controlled input (e.g. `x' OR '1'='1`) could break out of the literal and execute arbitrary SQL. Since `String` conforms to `XLExpression`, this affected every API accepting a runtime string: comparison operators, `printf`, `groupConcat(separator:)`, JSON paths, etc.

## Fix

Double embedded single quotes per the SQL standard in `XLiteFormatter.text(_:)`. The `StaticString` overload now routes through the same escaping, so compile-time literals containing apostrophes also encode correctly.

This is the minimal escaping fix. The structural fix — routing runtime values through the named-binding machinery by default — remains tracked as #30.

## Tests

Six new regression tests, all passing (suite: 224 tests, 0 failures):

**Syntax (`XLSyntaxTests`):**
- `O'Brien` → `'O''Brien'`
- `x' OR '1'='1` → fully escaped, cannot break out of the literal
- `a--b` preserved (no comment truncation inside a literal)
- empty string → `''`

**Execution round-trip (`XLExecutionTests`, real SQLite):**
- Insert + select a row keyed `O'Brien` — proves both the INSERT and WHERE paths escape correctly
- WHERE with an injection attempt matches zero rows instead of all rows

Fixes #99

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Tests/SQLTests/SQLSyntaxTests.swift [hunk 1]
```diff
@@ -57,6 +57,30 @@ final class XLSyntaxTests: XCTestCase {
...
+    func test_TextLiteral_PreservesCommentSequence() {
+        let expression: String = "a--b"
+        XCTAssertEqual(encoder.makeSQL(expression).sql, "'a--b'")
+    }
+
+
+    func test_TextLiteral_EmptyString() {
+        let expression: String = ""
+        XCTAssertEqual(encoder.makeSQL(expression).sql, "''")
+    }
```
[the remark this answers is not in the document]
Comment (lukevanin) [reply Tests/SQLTests/SQLSyntaxTests.swift:83]: Addressed in 6feee756:

- `test_TextLiteral_StaticString_EscapesSingleQuote` — exercises the `StaticString` overload directly on the formatter (`O'Brien` → `'O''Brien'`).
- `test_TextLiteral_EmbeddedNulCharacter` — characterization test for the `\0` edge case: a NUL cannot be escaped inside a SQL string literal, so the test verifies the quotes remain balanced and the value cannot break out of the literal.

Suite: 226 tests, 0 failures.

File: Sources/SwiftQL/SQLEncoder.swift [hunk 2]
```diff
@@ -240,11 +240,13 @@ public struct XLiteFormatter: XLFormatter {
     }
     
     public func text(_ text: String) -> String {
-        "'\(text)'"
+        // Embedded single quotes must be doubled per the SQL standard, otherwise
+        // the value breaks out of the literal (broken SQL at best, injection at worst).
+        "'\(text.replacingOccurrences(of: "'", with: "''"))'"
     }
```
[the remark this answers is not in the document]
Comment (lukevanin) [reply Sources/SwiftQL/SQLEncoder.swift:249]: Done in 530d2508 — added a `guard text.contains("'")` fast-path so the replacement (and its allocation) only runs when a quote is actually present. Suite still green (226 tests, 0 failures).

File: Tests/SQLTests/SQLSyntaxTests.swift [hunk 3]
```diff
@@ -57,6 +57,30 @@ final class XLSyntaxTests: XCTestCase {
...
         XCTAssertEqual(encoder.makeSQL(expression).sql, "'foo'")
     }
+
+
+    func test_TextLiteral_EscapesSingleQuote() {
+        let expression: String = "O'Brien"
+        XCTAssertEqual(encoder.makeSQL(expression).sql, "'O''Brien'")
+    }
+
+
```
[the remark this answers is not in the document]
Comment (lukevanin) [reply Tests/SQLTests/SQLSyntaxTests.swift:67]: This landed just before your review was submitted — `test_TextLiteral_StaticString_EscapesSingleQuote` (added in 6feee756) covers exactly this path: it calls the `StaticString` overload directly on the formatter and asserts `O'Brien` → `'O''Brien'`.
